ReactAndroid/src/androidTest/js/ScrollViewTestModule.js

```diff
@@ -37,54 +36,52 @@ class Item extends React.Component {
   }
 }
 
-var getInitialState = function() {
-  var data = [];
-  for (var i = 0; i < NUM_ITEMS; i++) {
-    data[i] = {text: 'Item ' + i + '!'};
+class ScrollViewTestApp extends React.Component {
+  constructor() {
+    super();
+    this.scrollView = React.createRef();
   }
-  return {
-    data: data,
-  };
-};
 
-var onScroll = function(e) {
-  ScrollListener.onScroll(
-    e.nativeEvent.contentOffset.x,
-    e.nativeEvent.contentOffset.y,
-  );
-};
+  getInitialState = () => {
+    var data = [];
+    for (var i = 0; i < NUM_ITEMS; i++) {
+      data[i] = {text: 'Item ' + i + '!'};
+    }
+    return {
+      data: data,
+    };
+  };
```

Comment:
Classes that extend `React.Component` do not call `getInitialState` to get their initial state. With the current configuration, `this.state` will be `null` inside the `render()` method, which will raise an exception. Instead of implementing `getinitialState`, you should create the state and assign it to `this.state` inside the constructor. Since we support class properties, we should just do this:

```JavaScript
class ScrollViewTestApp extends React.Component {
  scrollView = React.createRef();
  state = getInitialState();
```